src/providers/info.rs

```diff
@@ -113,6 +116,7 @@ pub struct DefaultInfoProvider<'p> {
     pub chain_id: &'p str,
     client: Arc<dyn HttpClient>,
     cache: Arc<dyn Cache>,
+    mainnet_cache: Arc<dyn Cache>,
```

Comment:
I don't think this should be required here. The Info provider always only operates on one chain. The chain id here is only for the purpose of generating the correct cache keys, right?

Reply:
Good catch. Removed it 👍